spec/support/filesystem_repo.rb

```diff
@@ -0,0 +1,51 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2014 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+def with_created_filesystem_repository(&block)
+  let(:repository) do
+    repo = FactoryGirl.build(:repository)
+
+    # ignoring the bugs on url as those are expected:
+    # 1) directory is not existing
+    # 2) configuration is not whitelisting the directory
+    if repo.valid? || (repo.errors.keys - [:url]).empty?
+      repo.save(validate: false)
+    else
+      repo.save!
+    end
+
+    repo
+  end
+
+  before do
+    allow(Setting).to receive(:enabled_scm).and_return(["Filesystem"])
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.